core/src/main/java/de/danoeh/antennapod/core/service/playback/LocalPSMP.java

```diff
@@ -952,7 +952,14 @@ public void onAudioFocusChange(final int focusChange) {
                 // Load next episode if previous episode was in the queue and if there
                 // is an episode in the queue left.
                 // Start playback immediately if continuous playback is enabled
-                nextMedia = callback.getNextInQueue(currentMedia);
+
+                // Repeat episode implementation
+                if (UserPreferences.repeatEpisode() && !wasSkipped) {
+                    nextMedia = currentMedia;
+                    nextMedia.setPosition(0);
+                } else {
+                    nextMedia = callback.getNextInQueue(currentMedia);
+                }
```

Comment:
I think this should be done outside LocalPSMP, but in `getNextInQueue`. The reason is that I want to reduce the dependence of the media players on the preferences and database. Also, it will then probably work on Chromecast.

Reply:
But LocalPSMP is already highly dependent on preferences. There are 17 uses of it. Are we going to get rid of those as well?

Reply:
> But LocalPSMP is already highly dependent on preferences. There are 17 uses of it. Are we going to get rid of those as well?

Oh right, sorry. I actually had `DBReader` in my mind when thinking about this.